src/touchscreen.c

```diff
@@ -0,0 +1,157 @@
+            if (hitboxes[i].circle.sentinel == TOUCHSCREEN_CIRCLE_SENTINEL) {
+                if (TouchscreenHitbox_PointIsInCircle(&hitboxes[i], gSystem.touchX, gSystem.touchY)) {
+                    return i;
+                }
+            } else {
+                if (TouchscreenHitbox_PointIsInRect(&hitboxes[i], gSystem.touchX, gSystem.touchY)) {
+                    return i;
+                }
+            }
+        }
+    }
+
+    return -1;
+}
+
+int TouchscreenHitbox_FindHitboxAtTouchNew(const TouchscreenHitbox *hitboxes) {
+    if (gSystem.touchNew) {
+        for (int i = 0; hitboxes[i].circle.sentinel != TOUCHSCREEN_RECTLIST_END; ++i) {
+            if (hitboxes[i].circle.sentinel == TOUCHSCREEN_CIRCLE_SENTINEL) {
+                if (TouchscreenHitbox_PointIsInCircle(&hitboxes[i], gSystem.touchX, gSystem.touchY)) {
+                    return i;
+                }
+            } else {
+                if (TouchscreenHitbox_PointIsInRect(&hitboxes[i], gSystem.touchX, gSystem.touchY)) {
+                    return i;
+                }
+            }
+        }
+    }
+
+    return -1;
+}
+
+BOOL TouchscreenHitbox_TouchHeldIsIn(const TouchscreenHitbox *hitbox) {
+    if (gSystem.touchHeld) {
+        if (hitbox->circle.sentinel == TOUCHSCREEN_CIRCLE_SENTINEL) {
+            return TouchscreenHitbox_PointIsInCircle(hitbox, gSystem.touchX, gSystem.touchY);
+        } else {
+            return TouchscreenHitbox_PointIsInRect(hitbox, gSystem.touchX, gSystem.touchY);
+        }
+    }
+
+    return FALSE;
+}
+
+BOOL TouchscreenHitbox_TouchNewIsIn(const TouchscreenHitbox *hitbox) {
+    if (gSystem.touchNew) {
+        if (hitbox->circle.sentinel == TOUCHSCREEN_CIRCLE_SENTINEL) {
+            return TouchscreenHitbox_PointIsInCircle(hitbox, gSystem.touchX, gSystem.touchY);
+        } else {
+            return TouchscreenHitbox_PointIsInRect(hitbox, gSystem.touchX, gSystem.touchY);
+        }
+    }
+
+    return FALSE;
+}
+
+BOOL System_GetTouchHeld(void) {
+    return gSystem.touchHeld;
+}
+
+BOOL System_GetTouchNew(void) {
+    return gSystem.touchNew;
+}
+
+BOOL System_GetTouchHeldCoords(u32 *x, u32 *y) {
+    if (gSystem.touchHeld) {
+        *x = gSystem.touchX;
+        *y = gSystem.touchY;
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+BOOL System_GetTouchNewCoords(u32 *x, u32 *y) {
+    if (gSystem.touchNew) {
+        *x = gSystem.touchX;
+        *y = gSystem.touchY;
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+int TouchscreenHitbox_FindHitboxAtPoint(const TouchscreenHitbox *hitboxes, u32 x, u32 y) {
+    for (int i = 0; hitboxes[i].circle.sentinel != TOUCHSCREEN_RECTLIST_END; ++i) {
+        if (hitboxes[i].circle.sentinel == TOUCHSCREEN_CIRCLE_SENTINEL) {
+            if (TouchscreenHitbox_PointIsInCircle(&hitboxes[i], x, y)) {
+                return i;
+            }
+        } else {
+            if (TouchscreenHitbox_PointIsInRect(&hitboxes[i], x, y)) {
```

Comment:
else if

Reply:
Strong disagree. Making this `else if` hides the similarity between these two blocks and hinders readability.

Reply:
precisely because they are similar is why I think it should be else if, one is circle, one is rect

Reply:
I'm sorry but your reasoning is self-contradictory.

Reply:
I mean they seem like two cases of the same check, therefore they should be grouped together